Update errors report design

guergana opened the pull request:
- fixes #634

guergana commented:
@romicolman @Faithkenny Maybe we should thing about some sort of indicator when the user has to scroll to the right? Sometimes the errors report has many columns and i think it's not immediately obvious that there is more information if you scroll to the right. I have made a gif to illustrate what I see:

![Peek 2024-11-15 19-34](https://github.com/user-attachments/assets/dece7af8-44c0-4dcb-8e76-f5b7028a2e4e)

Faithkenny commented:
> @romicolman @Faithkenny Maybe we should thing about some sort of indicator when the user has to scroll to the right? Sometimes the errors report has many columns and i think it's not immediately obvious that there is more information if you scroll to the right. I have made a gif to illustrate what I see:
> 
> ![Peek 2024-11-15 19-34](https://private-user-images.githubusercontent.com/3005239/386719756-dece7af8-44c0-4dcb-8e76-f5b7028a2e4e.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzE3NjQ2ODIsIm5iZiI6MTczMTc2NDM4MiwicGF0aCI6Ii8zMDA1MjM5LzM4NjcxOTc1Ni1kZWNlN2FmOC00NGMwLTRkY2ItOGU3Ni1mNWI3MDI4YTJlNGUuZ2lmP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTExNiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDExMTZUMTMzOTQyWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9N2I4NzhjY2UwY2FlYWE1MWJhYjFiYmY0YTEyOGQ5NDBjMzgyNGY2Mzc3ODcyN2UwNTdmMDM4OWEwYTI3NjEyMSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.9fLelpr8ZlLk2iAuhHQFcmcRs7H_WQNfimXAcegwlWg) [ ![Peek 2024-11-15 19-34](https://private-user-images.githubusercontent.com/3005239/386719756-dece7af8-44c0-4dcb-8e76-f5b7028a2e4e.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzE3NjQ2ODIsIm5iZiI6MTczMTc2NDM4MiwicGF0aCI6Ii8zMDA1MjM5LzM4NjcxOTc1Ni1kZWNlN2FmOC00NGMwLTRkY2ItOGU3Ni1mNWI3MDI4YTJlNGUuZ2lmP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTExNiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDExMTZUMTMzOTQyWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9N2I4NzhjY2UwY2FlYWE1MWJhYjFiYmY0YTEyOGQ5NDBjMzgyNGY2Mzc3ODcyN2UwNTdmMDM4OWEwYTI3NjEyMSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.9fLelpr8ZlLk2iAuhHQFcmcRs7H_WQNfimXAcegwlWg) ](https://private-user-images.githubusercontent.com/3005239/386719756-dece7af8-44c0-4dcb-8e76-f5b7028a2e4e.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzE3NjQ2ODIsIm5iZiI6MTczMTc2NDM4MiwicGF0aCI6Ii8zMDA1MjM5LzM4NjcxOTc1Ni1kZWNlN2FmOC00NGMwLTRkY2ItOGU3Ni1mNWI3MDI4YTJlNGUuZ2lmP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTExNiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDExMTZUMTMzOTQyWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9N2I4NzhjY2UwY2FlYWE1MWJhYjFiYmY0YTEyOGQ5NDBjMzgyNGY2Mzc3ODcyN2UwNTdmMDM4OWEwYTI3NjEyMSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.9fLelpr8ZlLk2iAuhHQFcmcRs7H_WQNfimXAcegwlWg) [ ](https://private-user-images.githubusercontent.com/3005239/386719756-dece7af8-44c0-4dcb-8e76-f5b7028a2e4e.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzE3NjQ2ODIsIm5iZiI6MTczMTc2NDM4MiwicGF0aCI6Ii8zMDA1MjM5LzM4NjcxOTc1Ni1kZWNlN2FmOC00NGMwLTRkY2ItOGU3Ni1mNWI3MDI4YTJlNGUuZ2lmP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTExNiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDExMTZUMTMzOTQyWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9N2I4NzhjY2UwY2FlYWE1MWJhYjFiYmY0YTEyOGQ5NDBjMzgyNGY2Mzc3ODcyN2UwNTdmMDM4OWEwYTI3NjEyMSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.9fLelpr8ZlLk2iAuhHQFcmcRs7H_WQNfimXAcegwlWg)

Very good catch @guergana. Let's review this on Monday. A max-width container is the first solution that jumps at me

romicolman commented:
Hi @guergana thanks for this implementation. To answer to your comments:

1) The scrolling problem: we are having a call today with Faith to address other elements and we added this to our discussion list.

2) During the user testing sessions, people did not detect the first column in the error tables as the row number. Faith has added a gray background on this first column to follow the Google Sheets and Excel logic and visually facilitates it detection. I checked this on the [Figma file](https://www.figma.com/design/cOpYjy35zB3dJf1GW1BC76/Open-Data-Editor?node-id=10002-64459&t=hHNUCVdrF7s2rGht-1) and it is there. Can you please add it?
3) In addition, can you please add the **Row number** title. In the current design, the row number is called **1**:

<img width="1149" alt="Captura de pantalla 2024-11-18 a la(s) 2 15 21 p  m" src="https://github.com/user-attachments/assets/d48c6bc8-09d3-4ef4-85b9-82823a800b26">

4) I also detected a problem when the content in the cell with the error is too large: we have a lot of blank space on the left menu and the width of the rows is disproportionate. Is there a way to solve this? Maybe by adding a tooltip for long content. This is a video showing the problem:

https://github.com/user-attachments/assets/068498ac-3e5c-494e-aab6-fd5338a20eee

5) In files with a lot of errors, when scrolling down the errors report to click show more, nothing happens. Here is a video showing the problem:


https://github.com/user-attachments/assets/6fa26ad8-732f-4a5f-abb9-70930aa74660

6) As we discussed on Slack we removed the detailed explanation of errors, to avoid repeating information. We will have more user testing sessions to check if this works if this change makes sense or if we need to include the element again.

Faithkenny commented:
On the call yesterday, I pointed out that we should have the **Settings section** (user guide & report an issue) in a **fixed position.** Giving more thought to this made me realize that I haven't shared a definition for the scrolling, and overscroll behaviour in the ODE. Here goes:

- The sidebar, control panel containing metadata and co. including the Pagination should be fixed always
- The Datagrid area which changes based on the viewed panel scrolls behind the control panel

This [**Prototype**](https://www.figma.com/proto/cOpYjy35zB3dJf1GW1BC76/Open-Data-Editor?page-id=10295%3A23001&node-id=10295-23002&node-type=frame&viewport=494%2C381%2C0.63&t=d0RzpM0XEYGm912G-1&scaling=contain&content-scaling=fixed) demonstrates what it should look like.

Happy to hear your thoughts @guergana @romicolman @pdelboca 

https://github.com/user-attachments/assets/fea82f21-5594-40e3-894a-aca8ae9e750d

guergana commented:
> Hi @guergana thanks for this implementation. To answer to your comments:
> 
>     1. The scrolling problem: we are having a call today with Faith to address other elements and we added this to our discussion list.
> 
>     2. During the user testing sessions, people did not detect the first column in the error tables as the row number. Faith has added a gray background on this first column to follow the Google Sheets and Excel logic and visually facilitates it detection. I checked this on the [Figma file](https://www.figma.com/design/cOpYjy35zB3dJf1GW1BC76/Open-Data-Editor?node-id=10002-64459&t=hHNUCVdrF7s2rGht-1) and it is there. Can you please add it?
> 
>     3. In addition, can you please add the **Row number** title. In the current design, the row number is called **1**:
> 
> 
> <img alt="Captura de pantalla 2024-11-18 a la(s) 2 15 21 p  m" width="1149" src="https://private-user-images.githubusercontent.com/20649846/387237332-d48c6bc8-09d3-4ef4-85b9-82823a800b26.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzIyMTU2ODcsIm5iZiI6MTczMjIxNTM4NywicGF0aCI6Ii8yMDY0OTg0Ni8zODcyMzczMzItZDQ4YzZiYzgtMDlkMy00ZWY0LTg1YjktODI4MjNhODAwYjI2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMjElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTIxVDE4NTYyN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWRkOWQ4OTMwN2I3ZjlhOTdhOGUxMzA3YTE1MWY4M2FmNDdmYzBhZDAwMzgyNDEzM2ZjNTgwOGIwZWE0ODVjOGEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.WqmJLKCnmMY7DvasEGxjjDguwibnPe7NPrLuIYeRm7w">
> 
>     4. I also detected a problem when the content in the cell with the error is too large: we have a lot of blank space on the left menu and the width of the rows is disproportionate. Is there a way to solve this? Maybe by adding a tooltip for long content. This is a video showing the problem:
> 
> 
> Grabacion.de.pantalla.2024-11-18.a.la.s.2.40.00.p.m.mov
> 
>     5. In files with a lot of errors, when scrolling down the errors report to click show more, nothing happens. Here is a video showing the problem:
> 
> 
> Grabacion.de.pantalla.2024-11-18.a.la.s.2.48.18.p.m.mov
> 
>     6. As we discussed on Slack we removed the detailed explanation of errors, to avoid repeating information. We will have more user testing sessions to check if this works if this change makes sense or if we need to include the element again.

Hello @romicolman could you please provide the file you used for testing? :pray:

romicolman commented:
Yes, here is [Distribucio.de.la.renda.P80_P20 (1).xlsx](https://github.com/user-attachments/files/17852929/Distribucio.de.la.renda.P80_P20.1.xlsx).

guergana commented:
> Yes, here is [Distribucio.de.la.renda.P80_P20 (1).xlsx](https://github.com/user-attachments/files/17852929/Distribucio.de.la.renda.P80_P20.1.xlsx).

@romicolman I have worked on the changes. . I couldn't yet find a solution to the tooltip in the table report when the cells are too big, I just set a fixed height for now and cut the rest out.

romicolman commented:
Hi @guergana! Just to post what we discussed yesterday. Bullet point 2 is the indication of the row number. Faith added a gray background to make that indication clearer to the user. Regarding the problem with the tooltip, I'll create a separate ticket. Let's merge this for now

romicolman reviewed (approved):
This is approved! You can merge it :)
